infra/lambda_functions/rest/createrevision/app.js

```diff
@@ -0,0 +1,41 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT-0
+
+const AWS = require("aws-sdk");
+
+const { DIAGRAMS_TABLE_NAME } = process.env;
+
+const ddb = new AWS.DynamoDB.DocumentClient({
+  apiVersion: "2012-08-10",
+  region: process.env.AWS_REGION,
+});
+
+exports.handler = async (event) => {
+  const body = JSON.parse(event.body);
+
+  const oldDiagramId = body.diagramData.diagramId;
+  const oldRootId = oldDiagramId.split("-")[0];
+  const newRevisionId = Date.now();
+  const newDiagramId = `${oldRootId}-${newRevisionId}`;
+
+  try {
+    await ddb
+      .put({
```

Comment:
Is the `put` function doing the sanitisation of the things in `Item` for us or do we have to do that? We're putting data directly from the body of the message into our DB...

Reply:
Here is an example of some issues with `scan`. Maybe `put` isn't susceptible to this type of attack: https://medium.com/appsecengineer/dynamodb-injection-1db99c2454ac

Reply:
This would probably also be helped by using typescript. That performs some sanity checks on the inputs e.g. 

```
interface MyObj {
    myString: string;
    myNumber: number;
}

let obj: MyObj = JSON.parse('{ "myString": "string", "myNumber": 4 }');
console.log(obj.myString);
console.log(obj.myNumber);
```